addons/mass_mailing/static/src/js/wysiwyg.js

```diff
@@ -18,17 +19,73 @@ const MassMailingWysiwyg = Wysiwyg.extend({
                 e.preventDefault();
             }
         });
+        this.snippetsMenuToolbar = this.toolbar;
         return res;
     },
 
     toggleLinkTools(options = {}) {
         this._super({
             ...options,
-            // Always open the dialog for the basic theme as it has no sidebar.
-            forceDialog: options.forceDialog || this.$iframeBody.find('.o_basic_theme').length
+            // Always open the dialog when the sidebar is folded.
+            forceDialog: options.forceDialog || this.snippetsMenu.folded
         });
+        if (this.snippetsMenu.folded) {
+            // Hide toolbar and avoid it being re-displayed after getDeepRange.
+            this.odooEditor.document.getSelection().collapseToEnd();
+        }
+    },
+
+    /**
+     * Sets SnippetsMenu fold state and switches toolbar.
+     * Instantiates a new floating Toolbar if needed.
+     *
+     * @param {Boolean} fold
+     */
+    setSnippetsMenuFolded: async function (fold = true) {
+        if (fold) {
+            this.snippetsMenu.setFolded(true);
+            if (!this.floatingToolbar) {
+                // Instantiate and configure new toolbar.
+                this.floatingToolbar = new Toolbar(this, 'web_editor.toolbar');
+                this.toolbar = this.floatingToolbar;
+                await this.toolbar.appendTo(document.createElement('void'));
+                this._configureToolbar({ snippets: false });
+                this._updateEditorUI();
+                this.setCSSVariables(this.toolbar.el);
+                this.odooEditor.setupToolbar(this.toolbar.el);
+                if (this.odooEditor.isMobile) {
+                    document.body.querySelector('.o_mail_body').prepend(this.toolbar.el);
+                } else {
+                    document.body.append(this.toolbar.el);
+                }
+            } else {
+                this.toolbar = this.floatingToolbar;
+            }
+            this.toolbar.el.classList.remove('d-none');
+            this.odooEditor.autohideToolbar = true;
+            this.odooEditor.toolbarHide();
+        } else {
+            this.snippetsMenu.setFolded(false);
+            this.toolbar = this.snippetsMenuToolbar;
+            this.odooEditor.autohideToolbar = false;
+            if (this.floatingToolbar) {
+                this.floatingToolbar.el.classList.add('d-none');
+            }
+        }
+        this.odooEditor.toolbar = this.toolbar.el;
     },
 
+    /**
+     * @override
+     */
+    openMediaDialog: function() {
+        this._super(...arguments);
+        if (this.snippetsMenu.folded) {
+            this.odooEditor.toolbarHide();
```

Comment:
It is not clear to me why this is needed.

Reply:
When everything is in the same document (ex: Note), opening the dialog triggers `_onSelectionChange`, which in turn updates the toolbar visility making it hidden. 
But when the wysiwyg is inside the iframe, opening the dialog (in the main document) does not trigger `_onSelectionChange`. As a result, we have the toolbar floating around, with the dialog opened.
These lines ensures the toolbar gets hidden when the media dialog is opened, making its behavior consistent with the one described above.

Reply:
Thank you ! 👍